script.js

```diff
@@ -0,0 +1,61 @@
+const getBooks = JSON.parse(localStorage.getItem('books'));
+let books = getBooks || [];
+
+const formText = document.querySelector('#articleForm');
+
+function displayData() {
+  let innerHTML = '';
+  for (let i = 0; i < books.length; i += 1) {
+    innerHTML += `
+            <article id="${books[i].length + 1}">
+                <div class="title">${books[i].title}</div>
+                <div class="author">${books[i].author}</div>
+                <button class="removeButton" onclick="removeData(${
+  books[i].id
+})">Remove</button>
+            </article>
+            <hr />
+            `;
+  }
+  document.querySelector('#articles').innerHTML = innerHTML;
+}
+
+formText.addEventListener('submit', (event) => {
+  event.preventDefault();
+  const titleText = formText.elements.titleText.value;
+  const authorText = formText.elements.authorText.value;
+  if (titleText && authorText) {
+    books.push({
+      id: books.length + 1,
+      title: titleText,
+      author: authorText,
+    });
+    const myBooks = JSON.stringify(books);
+    localStorage.setItem('books', myBooks);
+
+    displayData();
+    formText.elements.titleText.value = '';
+    formText.elements.authorText.value = '';
+  }
+});
+
+/* eslint-disable no-unused-vars */
+function removeData(id) {
+  books = books.filter((book) => id !== book.id);
+  const myBooks = JSON.stringify(books);
+  localStorage.setItem('books', myBooks);
+  displayData();
+}
```

Comment:
- [OPTIONAL] It is very nice that you are using an `id` for removing the books so it makes it easier to remove books with duplicate values, but the id of a book is the length of the array os books you have plus one, and after removing a book, if the books ids are not refreshed, it is possible to create new books and have books with same ids. It would be great if it is possible to refresh or to give a unique id for each book so this bug cannot happen.

Reply:
Thank you for the tip sir. we shall use that in the up comming challenges